### DISCUSSION TITLE
Change in behavior after 0.19.0

### DISCUSSION THREAD

**Discussion by alejandrogarin** [pointfreeco/swift-composable-architecture#682] [Show and tell]
I've been using TCA version 0.19.0 and after I updated I noticed a change rewarding reading state from side effects:

This test passes in 0.19.0 but fails in the latest version. 
I'm not sure if this is expected? Thanks!

```
func testCoalesceSynchronousActions_ReadingState() {
      var someSideEffect = {}
        
      let store = Store(
        initialState: 0,
        reducer: Reducer<Int, Int, () -> ()> { state, action, env in
          switch action {
          case 0:
              state = 100
            return .fireAndForget {
                env()
            }
          default:
            return .none
          }
        },
        environment: {
            someSideEffect()
        }
      )

      let viewStore = ViewStore(store)
      someSideEffect = {
          XCTAssertEqual(100, viewStore.state)
      }
      viewStore.send(0)
    }
```

**Comment by stephencelis** [comment]
This is due to one of the changes documented in [the release notes](https://github.com/pointfreeco/swift-composable-architecture/releases/tag/0.20.0), particularly:

> - Performance improvement/fix: a store publisher will only emit a single state change per synchronous `Store.send` and asynchronous effect received. This means synchronous effects (returned immediately from a reducer via `Effect.init(value:)`) will no longer result in extra publisher emissions. If your application previously depended on each of these emissions, use `Publisher.receive(on:)` to schedule these effects on the next run loop tick.

This means that if a synchronous effect has access to a view store, it will not have access to the state that is still being processed by the reducer. We would recommend _not_ injecting view stores into a store's environment, since there's a bit of a cyclical problem there: a view store depends on a store, which depends on an environment. If you make the environment also depend on the view store, you get into cycling territory.

Instead of accessing view store state in your environment/effect, you should always feed the current state from the reducer to whatever environment/effect needs it.

**Reply by alejandrogarin** [reply]
thanks !
